### PR TITLE
feat(auth): Added a User Auth configuration setting to disable/enable…

### PR DESCRIPTION
**Pull request by madhavilosetty-intel**
… MPS auth

[AB#7066](https://dev.azure.com/rbheBoards/c7d9d701-a85f-45ff-91f6-26c56ff2c4de/_workitems/edit/7066)
[AB#7075](https://dev.azure.com/rbheBoards/c7d9d701-a85f-45ff-91f6-26c56ff2c4de/_workitems/edit/7075)
[AB#7083](https://dev.azure.com/rbheBoards/c7d9d701-a85f-45ff-91f6-26c56ff2c4de/_workitems/edit/7083)

## PR Checklist

- [x] Unit Tests have been added for new changes
- [ ] API tests have been updated if applicable
- [x] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

### CODE REVIEW

File: .mpsrc [hunk 1]
```diff
@@ -25,6 +25,7 @@
   "db_provider": "postgres",
   "connection_string": "postgresql://<USERNAME>:<PASSWORD>@localhost:5432/mpsdb?sslmode=no-verify",
   "instance_name": "localhost",
+  "auth_enabled": true,
```
Comment (matt-primrose) [inline .mpsrc:28]: can we call this "web_auth_enabled" and move it next to "web_admin_user" so it is clear these are linked?

Comment (madhavilosetty-intel) [reply .mpsrc]: done

**Review by matt-primrose — DISMISSED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -56,8 +56,8 @@ function loadConfig (): configType {
   // build config object
   const config: configType = rc('mps')
 
-  if (!config.web_admin_password || !config.web_admin_user || !config.jwt_secret) {
-    logger.error('Web admin username, password and jwt secret are mandatory. Make sure to set values for these variables.')
+  if ((config.web_auth_enabled && (!config.web_admin_password || !config.web_admin_user)) || !config.jwt_secret) {
```
Comment (rsdmike) [inline src/index.ts:59]: I think to increase readability -- we can break this up a bit. To me, JWT and Web Auth are two separate pieces and shouldn't be checked together.
``` suggestion
if(config.web_auth_enabled){
    if(!config.web_admin_password || !config.web_admin_user){
//log error and exit
     }
}

if(!config.jwt_secret){
//log error and exit
}
```

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (madhavilosetty-intel) [reply src/index.ts]: done

**Review by rsdmike — APPROVED** [review]

**Pull request merged**